xla/service/gpu/fusions/transforms/simplify_arith.cc

```diff
@@ -228,6 +229,24 @@ struct RewriteTruncExtShuffle : public OpRewritePattern<mlir::gpu::ShuffleOp> {
   }
 };
 
+static std::optional<Interval> GetSelectRange(mlir::Operation* sel) {
+  // Match |x| implemented as (x >= 0) ? x : (0 - x).
+  auto x = mlir::matchers::m_Val(sel->getOperand(1));
+  if (!sel->getOperand(1).getType().isInteger(32) ||
```

Comment:
I guess we could also handle 64 bit int as well? We would just need to use std::numeric_limits<int64_t>::max() as upper bound, otherwise it should be the same. I think it is somewhat likely that s64 is used for indexing as well.

Reply:
I'd just go with isSignlessInteger and use `static_cast<int64_t>(1ull << (ty.getIntOrFloatBitWidth() - 1)) - 1` (the ull to prevent UB).